src/Emulator/Extensions/Mocks/HPSHostController.cs

```diff
@@ -30,11 +31,13 @@ public class HPSHostController : IHostMachineElement, IConnectable<II2CPeriphera
         public void AttachTo(II2CPeripheral obj)
         {
             currentSlave = obj;
+            currentDoubleSlave = (IDoubleWordPeripheral) obj;
```

Comment:
Why do you need a second instance variable, isn't `currentSlave` enough?

Reply:
No, they have different APIs unfortunately, even though the STM32F7 inherits both II2CPeripheral and IDoubleWordPeripheral, but it is unknown before the resc file is introduced, and we can't really use the instance directly.

Reply:
Instead of having an additional field, you can add a private property:
```
private IDoubleWordPeripheral CurrentDoubleSlave => currentSlave as IDoubleWordPeripheral;
```

That was you don't have to worry about clearing it on detach and any other inconsistencies between both fields.

Reply:
Good idea, thanks.